src/ripple/RippleEngine.ts

```diff
@@ -275,6 +303,155 @@ export class XrpEngine extends CurrencyEngine<
+      ? makeTokenId({
+          currency,
+          issuer
+        })
+      : undefined
+
+    const takerVal = isTakerToken ? value : String(takerAmount)
+
+    if (takerVal == null) {
+      this.error(
+        `parseRippleDexTxAmount: Transaction has token code ${currency} with no value`
+      )
+      return
+    }
+    const takerDenom =
+      tokenId == null
+        ? this.currencyInfo.denominations[0]
+        : this.builtinTokens[tokenId].denominations[0]
+    if (takerDenom == null) {
+      this.error(`parseRippleDexTxAmount: Unknown denom ${currency}`)
+      return
+    }
+    const nativeAmount = mul(takerVal, takerDenom.multiplier)
+
+    return {
+      nativeAmount,
+      pluginId: this.currencyInfo.pluginId,
+      tokenId
+    }
+  }
+
+  /**
+   * Parse potential DEX trades.
+   * Parse offer-related nodes to determine order status for saving to the
+   * EdgeTxAction
+   **/
+  processRippleDexTx = (
+    accountTx: AccountTxTransaction
+  ): EdgeTxActionSwap | undefined => {
+    const { meta, tx } = accountTx
+    if (tx == null || typeof meta !== 'object') return
+
+    const { AffectedNodes } = meta
+    const deletedNodes = AffectedNodes.filter(
+      node =>
+        isDeletedNode(node) && node.DeletedNode.LedgerEntryType === 'Offer'
+    ) as DeletedNode[]
+    const hasDeletedNodes = deletedNodes.length > 0
+    const hasModifiedNodes =
+      AffectedNodes.filter(
+        node =>
+          isModifiedNode(node) && node.ModifiedNode.LedgerEntryType === 'Offer'
+      ).length > 0
+    const createdNodes = AffectedNodes.filter(
+      node =>
+        isCreatedNode(node) && node.CreatedNode.LedgerEntryType === 'Offer'
+    )
+    // Shouldn't happen. Only possible to have one created node per order tx
+    if (createdNodes.length > 1) {
+      this.error('processRippleDexTx: OfferCreate: multiple created nodes')
+      return
+    }
+
+    let type: EdgeTxActionSwapType | undefined
+    let sourceAsset: EdgeAssetAmount | undefined
+    let destAsset: EdgeAssetAmount | undefined
+    // Any kind of limit order state - post (open & unfilled), partially
+    // filled, fully filled, but NOT canceled.
+    if (tx.TransactionType === 'OfferCreate') {
+      // Exactly one node was created. Order opened without any fills
+      const isOpenOrder = createdNodes.length === 1 // check modifiedNodes?
+
+      // Either an existing order that had partial fills, OR
+      // a new order that only matched exact offer amounts in the book
+      const isPartiallyFilled =
+        hasModifiedNodes || (isOpenOrder && hasDeletedNodes)
+
+      // Order was fully filled
+      const isFullyFilled = hasDeletedNodes && !isOpenOrder
+
+      // Don't care about partial fills - counting them as general fills
+      type =
+        isFullyFilled || isPartiallyFilled ? 'swapOrderFill' : 'swapOrderPost'
+
+      // Parse amounts
+      const { TakerPays, TakerGets } = tx
+      sourceAsset = this.parseRippleDexTxAmount(TakerGets)
```

Comment:
This always get the amounts from the `tx` object. But in the case of a partial order fulfillment, the correct amounts are in the Nodes.

Reply:
We don't actually display the amounts from here at this time. 

This implementation is tracking 2 things consistently across all order states: the state of the order, and the posted order amount. 

Fill amount is a completely separate thing and I think is a mistake to conflate into the `source/destAsset` amounts described here.

Also if we only track fill amount, we actually are mis-reporting later down the road when the order status changes. We won't have knowledge of that unless we go back and modify related transactions as we process newer transactions. Imo, fill amount should be a completely separate field.

At this point in time, the only thing we are certain of is the total amounts of the original offer. Even if we have a partial fill immediately upon posting, we don't know if a later tx from someone else further modified the fill amount.